pkg/proof/querier.go

```diff
@@ -123,7 +129,7 @@ func QueryShareInclusionProof(_ sdk.Context, path []string, req abci.RequestQuer
 
 // ParseNamespace validates the share range, checks if it only contains one namespace and returns
 // that namespace ID.
-func ParseNamespace(rawShares []shares.Share, startShare, endShare int) (appns.Namespace, error) {
+func ParseNamespace(rawShares []shares.Share, startShare int64, endShare int64) (appns.Namespace, error) {
```

Comment:
this is technically a breaking change

Reply:
Why are we switching here to `int64`? I don't think the shares numbers will go to reach `math.MaxInt`

Reply:
This removes the need to do additional range checking inside the body of this function.

Reply:
I am not sure if I see where this range check is happening, can you point the line please?